### PR TITLE
GOVUKAPP-781 User messaging for unverified config signature

### PR DESCRIPTION
**Pull request by TimGJD**
# Title
User messaging for unverified config signature

# Description
- Show forced update screen when app config signature is invalid on app launch.
- Align with iOS that any other app config errors result in app unavailable screen being shown on app launch.

## JIRA ticket(s)
  - [GOVAPP-781](https://govukverify.atlassian.net/browse/GOVUKAPP-781)

## Figma
  - [Figma board](https://www.figma.com/design/5u5cvVCgKY3Rng2ADNV30K/2024-07---GOV.UK-app---beta-designs?node-id=363-6730&node-type=section&t=3iXvvIdHgGeXyKG4-0)

### CODE REVIEW

File: app/src/main/kotlin/uk/govuk/app/AppViewModel.kt [hunk 1]
```diff
@@ -30,29 +31,46 @@ internal class AppViewModel @Inject constructor(
...
+                    shouldDisplayAppUnavailable = !flagRepo.isAppAvailable(),
+                    shouldDisplayForcedUpdate = flagRepo.isForcedUpdate(BuildConfig.VERSION_NAME),
+                    shouldDisplayRecommendUpdate = flagRepo.isRecommendUpdate(BuildConfig.VERSION_NAME),
+                    shouldDisplayAnalyticsConsent = analytics.isAnalyticsConsentRequired(),
+                    shouldDisplayOnboarding = flagRepo.isOnboardingEnabled() && !appRepo.isOnboardingCompleted(),
+                    shouldDisplayTopicSelection = flagRepo.isTopicsEnabled() && !appRepo.isTopicSelectionCompleted(),
+                    isSearchEnabled = flagRepo.isSearchEnabled(),
+                    isRecentActivityEnabled = flagRepo.isRecentActivityEnabled(),
+                    isTopicsEnabled = flagRepo.isTopicsEnabled()
+                )
```
Comment (gclssvglx) [inline app/src/main/kotlin/uk/govuk/app/AppViewModel.kt:53]: This is getting a bit of beast 🐻

**Review by gclssvglx — APPROVED** [review]

File: app/src/main/kotlin/uk/govuk/app/AppViewModel.kt (hunk 1 above)
Comment (gclssvglx) [reply app/src/main/kotlin/uk/govuk/app/AppViewModel.kt:53]: Not a blocker, just something we might want to look at if it needs to grow more.

Comment (TimGJD) [reply app/src/main/kotlin/uk/govuk/app/AppViewModel.kt]: Done (I hope is what you're expecting?)

**Review by gclssvglx — APPROVED** [review]
Looks a lot more flexible to me 👍

**Pull request merged**